File: GeneratedEntityReaderClasses/Components/CameraBoundComponent.cs
namespace GeneratedNoitaClasses;
public class CameraBoundComponent : NoitaComponentBase,  iNoitaType<CameraBoundComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool enabled { get; set; } = new();
	public NoitaFloat distance { get; set; } = new();
	public NoitaFloat distance_border { get; set; } = new();
	public NoitaInt max_count { get; set; } = new();
	public NoitaBool freeze_on_distance_kill { get; set; } = new();
	public NoitaBool freeze_on_max_count_kill { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		enabled.Read(s);
		distance.Read(s);
		distance_border.Read(s);
		max_count.Read(s);
		freeze_on_distance_kill.Read(s);
		freeze_on_max_count_kill.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		enabled.Write(s);
		distance.Write(s);
		distance_border.Write(s);
		max_count.Write(s);
		freeze_on_distance_kill.Write(s);
		freeze_on_max_count_kill.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/CameraBoundComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class CameraBoundComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    enabled:NoitaBool = field(default_factory=NoitaBool)
    distance:NoitaFloat = field(default_factory=NoitaFloat)
    distance_border:NoitaFloat = field(default_factory=NoitaFloat)
    max_count:NoitaInt = field(default_factory=NoitaInt)
    freeze_on_distance_kill:NoitaBool = field(default_factory=NoitaBool)
    freeze_on_max_count_kill:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.enabled.Read(s)
        self.distance.Read(s)
        self.distance_border.Read(s)
        self.max_count.Read(s)
        self.freeze_on_distance_kill.Read(s)
        self.freeze_on_max_count_kill.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.enabled.Write(s)
        self.distance.Write(s)
        self.distance_border.Write(s)
        self.max_count.Write(s)
        self.freeze_on_distance_kill.Write(s)
        self.freeze_on_max_count_kill.Write(s)

